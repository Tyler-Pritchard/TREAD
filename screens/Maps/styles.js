import { StyleSheet, Dimensions } from 'react-native';

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: 'black',
      borderWidth: 2,
      borderColor: 'red'
    },
    maps: {
      flex: 1,
      width: Dimensions.get('screen').width,
      height: Dimensions.get('screen').height,
      borderWidth: 2,
      borderColor: 'purple'
    },
    heatmap: {
      flex: 1,
      position: 'absolute',
      borderWidth: 2,
      borderColor: "orangered",
    }
  });

  export default styles;