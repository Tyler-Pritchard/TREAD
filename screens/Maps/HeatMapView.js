import React from 'react';
import WebView from 'react-native-webview';
import Heatmap from 'react-native-simpleheat';
import styles from './styles';

export default () => (


  <Heatmap
    style={styles.heatmap}
    WebView={WebView}
    region={{
      latitude: 37.797187, 
      longitude: -122.204921,
      latitudeDelta: 0.0625,
      longitudeDelta: 0.0125,
    }}
    data={[
      [
        -122.205930, // longitude
        37.805254, // latitude
        10, // intensity
      ],
    ]}
  />
);