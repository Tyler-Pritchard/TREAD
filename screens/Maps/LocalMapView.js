import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';
import {GOOGLE_API_KEY} from '../../config';
import { MapQuery } from '../components/SearchBar';
import MapView from 'react-native-maps';
import MapViewDirections from 'react-native-maps-directions';

const LocalMap = () => {

  const origin = {latitude: 37.797154, longitude: -122.204921};
  const destination = {latitude: 37.805570, longitude: -122.274335}

  return (
    <View style={styles.container}>

      <Text style={styles.text}>Welcome to Oakland</Text>
      <MapQuery />
      <MapView
        style={styles.maps}
        initialRegion={{
          latitude: 37.797154, 
          longitude: -122.204921,
          latitudeDelta: 0.0622,
          longitudeDelta: 0.0122,
        }
      }>
        <MapViewDirections 
          origin={origin}
          destination={destination}
          apikey={GOOGLE_API_KEY}
          strokeWidth={3}
          strokeColor="hotpink"
        />
      </MapView>

    </View>
  );
};

export default LocalMap;