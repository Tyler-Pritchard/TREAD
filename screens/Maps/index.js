import Maps from './MapsContainer';


export default Maps;