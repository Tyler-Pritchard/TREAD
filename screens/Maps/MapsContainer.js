import React from 'react';
import {View} from 'react-native';
import LocalMapView from './LocalMapView';

function Maps(props) {
    return (
        <View>
            <LocalMapView />
        </View>
    );
} 

export default Maps;