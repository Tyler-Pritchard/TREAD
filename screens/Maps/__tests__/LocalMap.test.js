import { render } from '@testing-library/react-native';
import Maps from '../index';

test('should verify container renders', () => {
    const {queryMaps} = render(<Maps {...props} />);
    const containerView = queryMaps('View');

    expect(viewContainer).toHaveLength(2)
});

// test.only('FIRST TEST THAT RUNS', () => {
//     expect(true).toBe(true);
//   });
  
//   test('LAST TEST WILL NOT RUN', () => {
//     expect('A').toBe('A');
//   });



//it renders default elements
//it renders marker1
//it handles valid input submission
//it renders marker2
//it renders polyline

// Given - some precondition
// When - some action executed by the function that youâ€™re testing
// Then - the expected outcome

//Add detox later for e2e and device simulation testing