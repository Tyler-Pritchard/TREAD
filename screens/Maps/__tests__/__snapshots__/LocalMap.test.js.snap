// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
